src/reconcile/rec_visibility.c

```diff
@@ -477,7 +477,8 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
         else if (select_tw->stop_ts != WT_TS_NONE || select_tw->stop_txn != WT_TXN_NONE) {
             /* If we only have a tombstone in the update list, we must have an ondisk value. */
             WT_ASSERT(session,
-              vpack != NULL && tombstone != NULL && !vpack->tw.prepare && last_upd->next == NULL);
+              vpack != NULL && tombstone != NULL && !vpack->tw.prepare && last_upd != NULL &&
+                last_upd->next == NULL);
```

Comment:
I can't understand what all the conditions here are meant to mean. The comment is informative but seems to only correspond to the first portion of the assertion. Please expand on the comment to make it more clear what the assertion is checking for.

I believe that the coverity failure is meaning that last_upd could technically be NULL coming through this path. If the assertion holds true it also looks like the code at line 509 below can't be reached. Should the code be updated?

Reply:
No, the code can be reached. last_upd->next must be null before calling rec_append_orig_value. It may be not null or still be null after calling rec_append_orig_value. I'll expand the comment to explain this.

Reply:
Coverity will not go away with the assert, we will just be validating our assumptions. We will have to mark the Coverity as intentional. This isn't the best I suppose, but a simple solution in this case.

Reply:
That's a sad. But we still can merge this change I think.

Reply:
Then we will see if that helps Coverity or not.

Reply:
Not related to the coverity issue, but really big assertions can be difficult to debug since there are so many possible things that could have triggered it and you don't know which it is without a core file. When it hits it could be anything. Could this be broken up into 3 assertions instead: 1. tombstone != NULL. 2. vpack != NULL && !vpack->tw.prepare. 3. last_upd != NULL && last_upd->next == NULL.

Reply:
Thanks Sue. I break the asserts.